stock_intercompany/models/stock_picking.py

```diff
@@ -0,0 +1,33 @@
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def create_counterpart_picking(self):
+        companies = self.env["res.company"].sudo().search([])
+        partners = {cp.partner_id: cp for cp in companies}
+        if self.partner_id in partners.keys():
```

Comment:
```suggestion
        if self.partner_id in partners:
```

the "in" operator searches on keys in a dictionary

Reply:
(!) Actually there is another issue here: I think that this code we will create a counterpart also if we purchase from another company, but the counterpart will be wrong, as it will be a reception picking on the other company (which is a supplier in this case).

For now, I suggest checking that the current picking is a shipping picking (for instance by checking that the location_dest_id of the picking is a Customers virtual location). If not, then the method should do nothing. 

THIS NEEDS FIXING.